### PR TITLE
search: set Zoekt and Searcher endpoints from job args

### PR DESCRIPTION
**Pull request by rvantonder**
We don't need to proxy via `args` when setting these values. The only reason we can't remove `args.Zoekt` and `args.SearcherURL` completely is because of repo has file dependencies. It's fine to continue because eventually repoHasFile will be the only thing depending on `args`.

## Test plan
Semantics-preserving

### CODE REVIEW

**Review by camdencheek — APPROVED** [review]

**Comment by rvantonder** [comment]
FYI @camdencheek I added another commit 6245e45fa0a620b62633d461965fefd1eda640f3  because I missed some more places where we can set the value directly from job args, and I realized I can set the args value only for repo/repoHasFile search block for it. Also running backend-integration to check my thinking.

File: internal/search/job/job.go [hunk 1]
```diff
@@ -336,10 +336,13 @@ func ToSearchJob(jargs *Args, q query.Q) (Job, error) {
 
 			if valid() {
 				if repoOptions, ok := addPatternAsRepoFilter(args.PatternInfo.Pattern, repoOptions); ok {
+					// Note: if we run a repo search,
+					// downstream logic relies the following
+					// args values to be set for repoHasFile
+					// operation. It is slated for removal.
```
Comment (camdencheek) [inline internal/search/job/job.go:342]: I appreciate that you maintained the weirdly short text wrapping

**Review by camdencheek — APPROVED** [review]
I do not revoke my approval

File: internal/search/job/job.go (hunk 1 above)
Comment (rvantonder) [reply internal/search/job/job.go]: meta-q

**Pull request merged**